app/Http/Controllers/VotesController.php

```diff
@@ -0,0 +1,208 @@
+            array(
+                'nominees' => $nominees->toArray()
+            )
+        );
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param Request $request
+     * @return Response
+     */
+    /**
+     * @OA\Post(
+     *   tags={"Votes"},
+     *   path="/api/votes",
+     *   summary="Creates a vote",
+     *     @OA\RequestBody(
+     *         @OA\MediaType(
+     *             mediaType="application/json",
+     *             @OA\Schema(
+     *                 @OA\Property(
+     *                     property="id",
+     *                     type="integer"
+     *                 ),
+     *                 @OA\Property(
+     *                     property="email",
+     *                     type="string"
+     *                 ),
+     *                 @OA\Property(
+     *                     property="consent",
+     *                     type="integer"
+     *                 example={"id": 11, "email" : "tomes@utb.cz", "consent" : 1}
+     *             )
+     *         )
+     *     ),
+     *   @OA\Response(response=200, description="OK"),
+     *   @OA\Response(response=401, description="Unauthorized"),
+     *   @OA\Response(response=404, description="Not Found")
+     * )
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+            'id' => ['required'],
+            'email' => ['required'],
+            'consent' => ['required']
+        ]);
+
+        if ((int) $request->input('consent') === 0) {
+            return response()->json([
+                'error' => 'Consent must be accepted!'
+            ], 400);
+        }
+
+        $emailValidation = $this->validateEmailExistsInDb($request->input('email'));
+
+        if ($emailValidation === true) {
+            return response()->json([
+                'error' => 'This email address has already been used!'
+            ], 400);
+        }
+
+        $vote = Vote::create([
+            'nominee_id' => $request->input('id'),
+            'voter_email' => $request->input('email'),
+            'consent' => (int) $request->input('consent')
+        ]);
+
+        $data = ['vote' => $vote];
+        return response()->json($data, 200);
+    }
+
+    function validateEmailExistsInDb($email): bool
+    {
+        $votes = Vote::all();
+
+        if ($votes->contains('voter_email', $email))
+            return true;
+        else
+            return false;
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id
+     * @return Response
+     */
+    public function showNominee($id)
+    {
+        return Nominee::find($id);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param Request $request
+     * @param int $id
+     * @return Response
+     */
+    public function updateNominee(Request $request, $id)
+    {
+        $nominee = Nominee::find($id);
+        $nominee->update($request->all());
```

Comment:
Pozor na name fieldu z POSTu aby ti to proslo

Reply:
Well, tohle vlastně ani nikde nepoužívám